src/clients/SpokePoolClient/SVMSpokePoolClient.ts

```diff
@@ -231,6 +233,41 @@ export class SVMSpokePoolClient extends SpokePoolClient {
     };
   }
 
+  public override async findAllDeposits(depositId: BigNumber): Promise<MultipleDepositSearchResult> {
+    // TODO: Should we have something like this? In findDeposit we don't look in memory.
+    // // First check memory for deposits
+    // const memoryDeposits = this.getDepositsForDepositId(depositId);
+    // if (memoryDeposits.length > 0) {
+    //   return { found: true, deposits: memoryDeposits };
+    // }
+
+    // If no deposits found in memory, try to find on-chain
+    const deposits = await findAllDeposits(this.svmEventsClient, depositId);
+    if (!deposits || deposits.length === 0) {
+      return {
+        found: false,
+        code: InvalidFill.DepositIdNotFound,
+        reason: `Deposit with ID ${depositId} not found`,
+      };
+    }
+
+    // Enrich all deposits with additional information
+    const enrichedDeposits = await Promise.all(
+      deposits.map(async (deposit: DepositWithBlock) => ({
+        ...deposit,
+        quoteBlockNumber: await this.getBlockNumber(Number(deposit.quoteTimestamp)),
```

Comment:
One comment - `quoteBlockNumber` should be a mainnet block number, typically resolved via the HubPoolClient. You should be able to find examples of this elsewhere in the SpokePoolClient.